### ISSUE TITLE
[Types] Make `queryParams` strongly-typed

### ISSUE THREAD

**Issue by sewera** [pocketbase/js-sdk#101]
Right now, there are no types for `queryParams`; the default value is `{}`, so any object will match this type.

Some autocomplete will be nice, so I propose such alternative:

```ts
type GetListQueryParams = {
  page?: number
  perPage?: number
  sort?: string
  filter?: string
  expand?: string
}

// BaseCrudService@41
protected _getList<T = M>(basePath: string, page = 1, perPage = 30, queryParams: GetListQueryParams = {}): Promise<ListResult<T>> {
// ...
}
```

**Comment by sewera** [comment]
If it's okay with you, I can make those changes and open a PR.

I can also merge the `BaseCrudService` and `CrudService` (resolve [this todo](https://github.com/pocketbase/js-sdk/blob/dd17b2dd25b22df1dbb73cdbdb49ff3a429fbd44/src/services/utils/BaseCrudService.ts#L6)) if you want me to :)

**Comment by ganigeorgiev** [comment]
Hm, this was requested previously as part of https://github.com/pocketbase/js-sdk/issues/64.

For now let's keep the scope of the issue smaller and not change the `BaseCrudService` and `CrudService` since this will require some major updates to the tests too.

We can add type for the query params but the declaration(s) needs to be a little more "relaxed" for better backward comparability and to allow setting also custom query props, eg. something like:

```ts
export interface BaseQueryParams {
    [key: string]: any;

    expand?:      string;
    $autoCancel?: boolean;
    $cancelKey?:  string;
}

export interface ListQueryParams extends BaseQueryParams {
    page?:    number;
    perPage?: number;
    sort?:    string;
    filter?:  string;
}
```

_@sewera If you don't have the time to work on this, I'll implement it sometime at the end of the week._

**Comment by sewera** [comment]
I'll try to put together a PR with those changes by tomorrow.

**Comment by sewera** [comment]
@ganigeorgiev, please find PR #102 ready for review.

**Comment by ganigeorgiev** [comment]
The following types were added in master:

```js
  BaseQueryParams
  ListQueryParams
  RecordQueryParams
  RecordListQueryParams
  LogStatsQueryParams
  FileQueryParams
```

They will be available with the v0.8.4 release sometime later this week.
